base/pkg/entry.jl

```diff
@@ -553,6 +553,20 @@ function resolve(
         end
         rethrow(err)
     end
+    imported = String[]
+    for (pkg,(ver1,ver2)) in changed
```

Comment:
Rather than having a second loop for this, I would just update the `imported` list in the `for (pkg,(ver1,ver2)) in changes` loop.